Remove HAVE_OS_DARK_MODE_SUPPORT

nt1m opened the pull request:
#### c3d57c8fad33b07b852f4cd4cbc4cdfd892cbf91
<pre>
Remove HAVE_OS_DARK_MODE_SUPPORT
<a href="https://bugs.webkit.org/show_bug.cgi?id=284482">https://bugs.webkit.org/show_bug.cgi?id=284482</a>
<a href="https://rdar.apple.com/141309130">rdar://141309130</a>

Reviewed by Aditya Keerthi.

This is enabled everywhere. Remove it so code becomes easier to read.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/html.css:
(@namespace &quot;<a href="http://www.w3.org/1999/xhtml&quot">http://www.w3.org/1999/xhtml&quot</a>;;):
(#if defined(WTF_PLATFORM_IOS_FAMILY) &amp;&amp; WTF_PLATFORM_IOS_FAMILY):
(#endif):
(#endif // defined(ENABLE_INPUT_TYPE_COLOR) &amp;&amp; ENABLE_INPUT_TYPE_COLOR):
(#if defined(HAVE_OS_DARK_MODE_SUPPORT) &amp;&amp; HAVE_OS_DARK_MODE_SUPPORT): Deleted.
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.h:
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::overridePrefersColorScheme):
(WebCore::InspectorPageAgent::defaultUserPreferencesDidChange):
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::updateBackgroundRecursively):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::setUseDarkAppearanceOverride):
* Source/WebCore/platform/graphics/GraphicsContext.h:
(WebCore::GraphicsContext::setUseDarkAppearance):
* Source/WebCore/platform/graphics/GraphicsContextState.cpp:
(WebCore::GraphicsContextState::mergeSingleChange):
(WebCore::GraphicsContextState::mergeAllChanges):
(WebCore::stateChangeName):
(WebCore::GraphicsContextState::dump const):
* Source/WebCore/platform/graphics/GraphicsContextState.h:
(WebCore::GraphicsContextState::setUseDarkAppearance):
* Source/WebCore/rendering/TextPaintStyle.cpp:
(WebCore::TextPaintStyle::operator== const):
(WebCore::computeTextPaintStyle):
(WebCore::updateGraphicsContext):
* Source/WebCore/rendering/TextPaintStyle.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::systemColor const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:
* Tools/TestWebKitAPI/Tests/WebKit/mac/ForceLightAppearanceInBundle.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CopyRTF.mm:
(TEST(CopyRTF, StripsDefaultTextColorOfDarkContent)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteHTML.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteRTFD.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewUnderPageBackgroundColor.mm:
(defaultBackgroundColor):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView forceDarkMode]):

Canonical link: <a href="https://commits.webkit.org/287712@main">https://commits.webkit.org/287712@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0027f56ac7b2964102d213d5cbfd2d78966aeee2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80590 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59597 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34523 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85114 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31572 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82701 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68658 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7903 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/62955 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/20757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83659 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/53071 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73369 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43260 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/50364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/27529 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30031 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/73570 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/71510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28052 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86547 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/79649 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7816 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/5515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/71250 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7991 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69206 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/70490 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14499 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13446 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/102056 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12479 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7778 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/13297 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/24846 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7617 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/11136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9422 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/Versions/Inspector-iOS-14.0.json:
```diff
@@ -3639,7 +3639,7 @@
             "id": "Appearance",
             "type": "string",
-            "condition": "(defined(ENABLE_DARK_MODE_CSS) && ENABLE_DARK_MODE_CSS) || (defined(HAVE_OS_DARK_MODE_SUPPORT) && HAVE_OS_DARK_MODE_SUPPORT)",
+            "condition": "(defined(ENABLE_DARK_MODE_CSS) && ENABLE_DARK_MODE_CSS)",
```
everything inside `Source/WebInspectorUI/Versions` exists for backwards compatibility and being able to look up the state of things in the past. so please do not edit them

pxlcoder replied:
ooh, good to know! might be good to have a style checker rule for that at some point

nt1m committed:
Remove HAVE_OS_DARK_MODE_SUPPORT
https://bugs.webkit.org/show_bug.cgi?id=284482
rdar://141309130

Reviewed by Aditya Keerthi.

This is enabled everywhere. Remove it so code becomes easier to read.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/html.css:
(@namespace "http://www.w3.org/1999/xhtml";):
(#if defined(WTF_PLATFORM_IOS_FAMILY) && WTF_PLATFORM_IOS_FAMILY):
(#endif):
(#endif // defined(ENABLE_INPUT_TYPE_COLOR) && ENABLE_INPUT_TYPE_COLOR):
(#if defined(HAVE_OS_DARK_MODE_SUPPORT) && HAVE_OS_DARK_MODE_SUPPORT): Deleted.
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.h:
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::overridePrefersColorScheme):
(WebCore::InspectorPageAgent::defaultUserPreferencesDidChange):
* Source/WebCore/inspector/agents/InspectorPageAgent.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::updateBackgroundRecursively):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::setUseDarkAppearanceOverride):
* Source/WebCore/platform/graphics/GraphicsContext.h:
(WebCore::GraphicsContext::setUseDarkAppearance):
* Source/WebCore/platform/graphics/GraphicsContextState.cpp:
(WebCore::GraphicsContextState::mergeSingleChange):
(WebCore::GraphicsContextState::mergeAllChanges):
(WebCore::stateChangeName):
(WebCore::GraphicsContextState::dump const):
* Source/WebCore/platform/graphics/GraphicsContextState.h:
(WebCore::GraphicsContextState::setUseDarkAppearance):
* Source/WebCore/rendering/TextPaintStyle.cpp:
(WebCore::TextPaintStyle::operator== const):
(WebCore::computeTextPaintStyle):
(WebCore::updateGraphicsContext):
* Source/WebCore/rendering/TextPaintStyle.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::systemColor const):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:
* Tools/TestWebKitAPI/Tests/WebKit/mac/ForceLightAppearanceInBundle.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CopyRTF.mm:
(TEST(CopyRTF, StripsDefaultTextColorOfDarkContent)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteHTML.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PasteRTFD.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewUnderPageBackgroundColor.mm:
(defaultBackgroundColor):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView forceDarkMode]):

Canonical link: https://commits.webkit.org/287712@main
